internal/node/machine_mgr.go

```diff
@@ -547,12 +551,46 @@ func (m *MachineManager) cleanSockets() {
 
 func (m *MachineManager) generateTriggerHandler(vm *runningFirecracker, tsub string, request *agentapi.DeployRequest) func(msg *nats.Msg) {
 	return func(msg *nats.Msg) {
+		ctx := m.ctx
+
+		ctx, parentSpan := tracer.Start(
+			ctx,
```

Comment:
All the `ctx` floating around are kind of difficult to read. Can we use `m.ctx` here directly?

Reply:
no, we can't. We produce a new context every time we start a new span.